code/modules/antagonists/vampire/vampire.dm

```diff
@@ -0,0 +1,288 @@
+		/obj/effect/proc_holder/spell/vampire/self/cloak = 150,
+		/obj/effect/proc_holder/spell/vampire/targetted/disease = 150,
+		/obj/effect/proc_holder/spell/vampire/bats = 200,
+		/obj/effect/proc_holder/spell/vampire/self/screech = 200,
+		/datum/vampire_passive/regen = 200,
+		/obj/effect/proc_holder/spell/vampire/shadowstep = 250,
+		/obj/effect/proc_holder/spell/vampire/self/jaunt = 300,
+		/obj/effect/proc_holder/spell/vampire/targetted/enthrall = 300,
+		/datum/vampire_passive/full = 500)
+
+/datum/antagonist/vampire/on_gain()
+	give_objectives()
+	check_vampire_upgrade()
+	ticker.mode.vampires += owner
+	owner.special_role = "vampire"
+	owner.current.faction += "vampire"
+	ticker.mode.update_vampire_icons_added(owner)
+	..()
+
+/datum/antagonist/vampire/on_removal()
+	remove_vampire_powers()
+	owner.current.faction -= "vampire"
+	ticker.mode.vampires -= owner
+	owner.special_role = null
+	owner.current.create_attack_log("<span class='danger'>De-vampired</span>")
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/hud/hud = H.hud_used
+		if(hud.vampire_blood_display)
+			hud.remove_vampire_hud()
+	ticker.mode.update_vampire_icons_removed(owner)
+	for(var/O in objectives_given)
+		owner.objectives -= O
+	LAZYCLEARLIST(objectives_given)
+	if(owner.current)
+		if(issilicon(owner.current))
+			to_chat(owner.current, "<span class='userdanger'>You have been turned into a robot! You can feel your powers fading away...</span>")
+		else
+			to_chat(owner.current,"<span class='userdanger'>Your powers have been quenched! You are no longer a vampire</span>")
+	owner.special_role = null
+	..()
+
+/datum/antagonist/vampire/greet()
+	to_chat(owner.current, "<span class='userdanger'>You are a Vampire!</span>")
+	to_chat(owner.current, "<span class='danger'>To bite someone, target the head and use harm intent with an empty hand. Drink blood to gain new powers.</span>")
+	to_chat(owner.current, "<span class='danger'>You are weak to holy things and starlight. Don't go into space and avoid the Chaplain, the chapel and especially Holy Water.</span>")
+	if(owner.current.mind.assigned_role == "Clown")
+		to_chat(owner.current, "Your lust for blood has allowed you to overcome your clumsy nature allowing you to wield weapons without harming yourself.")
+		owner.current.mutations.Remove(CLUMSY)
+
+	if(LAZYLEN(objectives_given))
+		owner.announce_objectives()
+
+/datum/antagonist/vampire/proc/give_objectives()
+	var/datum/objective/blood/blood_objective = new
+	blood_objective.owner = owner
+	blood_objective.gen_amount_goal(150, 400)
+	add_objective(blood_objective)
+
+	var/datum/objective/assassinate/kill_objective = new
+	kill_objective.owner = owner
+	kill_objective.find_target()
+	add_objective(kill_objective)
+
+	var/datum/objective/steal/steal_objective = new
+	steal_objective.owner = owner
+	steal_objective.find_target()
+	add_objective(steal_objective)
+
+
+	switch(rand(1,100))
```

Comment:
Interesting implementation of a `switch()` just for this. I would have thought just an `if(prob(80)) else` would have been the first port of call.

Reply:
Agreed, `prob` would make more sense here.